doc/ipythonconsole.rst

```diff
@@ -2,91 +2,145 @@
+
+.. image:: images/console/console-options-menu.png
+   :alt: Spyder IPython Console with options menu
+
+You can also change the name of the current :guilabel:`IPython console` tab by simply double-clicking it, or with the :guilabel:`Rename tab` option.
+
+.. image:: images/console/console-rename.gif
+   :alt: Spyder IPython Console showing renaming console
+
+
+
+.. _connecting-external-kernel:
+
+==================================
+Using external kernels with Spyder
+==================================
+
+You can connect to external local and remote kernels (including those managed by QtConsole sessions or the Jupyter Notebook) through the :guilabel:`Connect to an existing kernel` dialog under the :guilabel:`Consoles` menu, the pane menu or its context menu.
+External kernels still support all of the :ref:`features<console-features>` of Spyder's IPython console.
+To work with Spyder, a compatible version of the ``spyder-kernels`` package :ref:`must be installed <starting-kernel-problems-ref>` in the environment or machine in which the external kernel is running.
+
+.. image:: images/console/console-menu.png
+   :alt: Connect to external kernel dialog of the Spyder IPython console
 
-#. Copy the connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on (if remote) or note its location (if local).
+
+Connect to a local kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To connect to a local kernel that is already running (e.g. under Jupyter notebook),
+
+#. Note the location of the kernel connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) of the kernel you are running, or its process ID (PID).
 
    You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   Usually, the connection file you are looking for will be one of the newest in this directory, corresponding to the time you started the external kernel.
 
-#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Console` menu or the :guilabel:`IPython Console` pane's "Gear" menu.
+#. In Spyder, click :guilabel:`Connect to an existing kernel` from the :guilabel:`Console` menu or the :guilabel:`IPython Console` pane's "Gear" menu.
 
-#. Browse for or enter the path to the connection file from the previous step.
-   If you're connecting to a local kernel, click :guilabel:`Ok` and Spyder should connect to the kernel; if a remote kernel, proceed to the final step.
+#. Browse for or enter the path to the :guilabel:`Connection file` from the previous step.
 
-   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to :file:`{jupyter/runtime/dir/path}/kernal-{id}.json` on your local machine.
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to the full path to the file, :file:`{jupyter/runtime/dir/path}/kernal-{id}.json`.
```

Comment:
```suggestion
   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to the full path of the file, :file:`{jupyter/runtime/dir/path}/kernal-{id}.json`.
```

I think this needs to be `of` instead of `to` (not really sure).

Reply:
"to" is also technically correct, but is repetitive so I agree "of" is better